atlas/packages/app/src/shared/components/Button/Button.style.tsx

```diff
@@ -0,0 +1,188 @@
+import React from 'react'
+import { css } from '@emotion/core'
+import styled from '@emotion/styled'
+import { BlockIcon } from '@/shared/icons'
+import { typography, colors, spacing } from '../../theme'
+
+export type ButtonStyleProps = Partial<{
+  variant: 'primary' | 'secondary' | 'tertiary'
+  full: boolean
+  size: 'regular' | 'small' | 'smaller'
+  children: React.ReactNode
```

Comment:
Children are needed when calculating the size, onClick is not actually needed since there is clickable available.

Reply:
All right, I've missed that. I'm tempted to ask however, do you see a situation in which we would use a button without any content?

Reply:
Genuinely curious, I don't think I've seen a situation like this before

Reply:
After thinking about your remark I've looked at the code again and found a bug.
`StyledButton` used `childrens` to know how to set the padding (to make Icon only buttons squared), but always caught the `Icon` itself since that was a child of the `StyledButton` component. 
I've added a `hasText` prop which is true if the button has text and false otherwise, this way Icon buttons are now squared.